PyEMD/CEEMDAN.py

```diff
@@ -319,6 +324,16 @@ def emd(self, S, T, max_imf=-1):
         """
         return self.EMD.emd(S, T, max_imf)
 
+    def get_imfs_and_residue(self):
+        """
+        Provides access to separated imfs and residue from recently analysed signal.
+        :return: (imfs, residue)
+        """
+        if self.C_IMF is None or self.residue is None:
+            raise ValueError('No IMF found. Please, run EMD method or its variant first.')
+        else:
+            return self.C_IMF, self.residue
```

Comment:
I think more Pythonic way would to simple return whatever is available without raising exceptions. In the greenpath scenario there's no difference and in the alternative the user still needs to check the result.

However, it might help to debug in case of issues so that's an added value.

Reply:
Hi @laszukdawid 
Yes, the more Pythonic way is to use try...except statements, but particularly in **emd.py** the `self.imfs` and `self.residue` were declared in the __init__ constructor. For the other EMD variants (EEMD and CEEMDAN), their IMFs and residue were declared in a class method outside the constructor, so it actually raises an AttributeError. Please, see my comment below where you suggested to include a test for the exception.